components/org.wso2.carbon.identity.api.user.authorized.apps/org.wso2.carbon.identity.rest.api.user.authorized.apps.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/authorized/apps/v1/factories/AuthorizedAppsServiceFactory.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.rest.api.user.authorized.apps.v1.factories;
+
+import org.wso2.carbon.identity.oauth.OAuthAdminServiceImpl;
+import org.wso2.carbon.identity.rest.api.user.authorized.apps.v1.core.AuthorizedAppsService;
+import org.wso2.carbon.identity.rest.api.user.authorized.apps.v1.utils.AuthorizedAppsServicesHolder;
+
+/**
+ * This class is responsible for initializing AuthorizedAppsService.
+ */
+public class AuthorizedAppsServiceFactory {
+
+    private AuthorizedAppsServiceFactory() {
+
+    }
+
+    private static class AuthorizedAppsServiceHolder {
```

Comment:
Any specific reason to adding sevice holder here? Can't we use the same approach used in [1]
[1] https://github.com/wso2/identity-api-user/pull/206/files#diff-148b635787c4b5c7ce0d1d77b45c189df0541aa5ac5a9ed4e5c093523b41b7e6R29-R57